sdk/src/main/scala/io/horizen/account/state/ForgerStakeV2MsgProcessor.scala

```diff
@@ -185,10 +186,18 @@ object ForgerStakeV2MsgProcessor extends NativeSmartContractWithFork  with Forge
     Array.emptyByteArray
   }
 
-  def doUpdateForger(invocation: Invocation, gasView: BaseAccountStateView): Array[Byte] = {
+  def doUpdateForger(invocation: Invocation, gasView: BaseAccountStateView, context: ExecutionContext): Array[Byte] = {
     requireIsNotPayable(invocation)
     checkForgerStakesV2IsActive(gasView)
 
+    val currentEpoch = context.blockContext.consensusEpochNumber
+    val activationEpoch = Version1_4_0Fork.get(currentEpoch).mainnetActivationEpoch
+    if (currentEpoch < (activationEpoch + NUM_OF_EPOCHS_AFTER_FORK_ACTIVATION_FOR_UPDATE_FORGER) ) {
```

Comment:
Won't this cause a sync failure in Pregobi? I think we did an updateForger on Pregobi before 2 epochs passed after activation. In addition, shouldn't be used a different activationEpoch depending on the type of network (regtest, testne, mainnet)?

Reply:
see my commet in Version1_4_0Fork.scala : with that change, we should have already the correct value for the current running network